util/include/sign_verify_utils.hpp

```diff
@@ -0,0 +1,33 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "openssl_utils.hpp"
+#include "crypto/eddsa/EdDSA.h"
+#include "crypto/eddsa/EdDSASigner.hpp"
+#include "crypto/eddsa/EdDSAVerifier.hpp"
+
+namespace concord::signerverifier {
+#ifdef USE_CRYPTOPP_RSA
```

Comment:
you said that the signature algorithm needed to be determined on runtime, where does that happen, and why do you need this ifdef if this is the case?

Reply:
Replaced with factory class implementation which is available in feature branch: https://github.com/arc-vmware/concord-bft/blob/crypto-decoupling-openssl-lib/util/include/crypto/factory.hpp